joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/astcreation/FullNameProvider.scala

```diff
@@ -327,24 +347,16 @@ trait FullNameProvider { this: AstCreator =>
     s"${fullName(enumSpecifier.getParent)}.${ASTStringUtil.getSimpleName(enumSpecifier.getName)}"
   }
 
-  private def fullNameForIASTFunctionDeclarator(f: IASTFunctionDeclarator): String = {
-    if (f.getParent.isInstanceOf[IASTFunctionDefinition]) {
-      s"${fullName(f.getParent)}"
-    } else {
-      s"${fullName(f.getParent)}.${shortName(f)}"
-    }
+  private def fullNameForIASTElaboratedTypeSpecifier(e: IASTElaboratedTypeSpecifier): String = {
+    s"${fullName(e.getParent)}.${ASTStringUtil.getSimpleName(e.getName)}"
   }
 
-  private def fullNameForIASTFunctionDefinition(f: IASTFunctionDefinition): String = {
-    if (f.getDeclarator != null) {
-      ASTStringUtil.getQualifiedName(f.getDeclarator.getName)
-    } else {
-      s"${fullName(f.getParent)}.${shortName(f)}"
-    }
+  private def fullNameForIASTFunctionDeclarator(f: IASTFunctionDeclarator): String = {
+    Try(fixQualifiedName(ASTStringUtil.getQualifiedName(f.getName))).getOrElse(nextClosureName())
```

Comment:
Instead, we use the qualified name API. The try is there just in case getName or getDeclerator return null.